chain/ethevents/events.go

```diff
@@ -127,7 +133,108 @@ func NewEthEvents(contractsAddresses []common.Address, signer *ethereum.SignKeys
 			eventQueue:            make(map[string]*logEvent),
 			eventProcessorRunning: false,
 		},
-	}, nil
+	}
+
+	if scrutinizer != nil {
+		log.Infof("starting ethevents with scrutinizer enabled")
+		ethev.Scrutinizer = scrutinizer
+		ethev.Scrutinizer.AddEventListener(ethev)
+	}
+
+	return ethev, nil
+}
+
+// OnComputeResults is called once a process result is computed by the scrutinizer.
+func (ev *EthereumEvents) OnComputeResults(results *models.ProcessResult) {
+	tctx, cancel := context.WithTimeout(context.Background(), time.Minute*5)
```

Comment:
it would be more idiomatic if this func received a parent ctx too, then the parent can set the timeout to whatever it wants in the call site

Reply:
This is a bit tricky. OnComputeResults is called by checkFinishedProcesses at a given height, the height is determined by the Vochain itself and is autoincremental on each commit which happens on every consensus round (atm 10s). I think in this case and given your comment is it better to remove this context and figure out how can we tie the commit time with the results processing timeout.
Another reason is that the defined interface on the Vochain does not accept context and adding the mentioned context will require to make more changes on all the other event listener methods. We can discuss more in depth :)

Reply:
gotcha, then maybe add TODOs